### PR TITLE
feature: IP lookup

### LINKED ISSUE #1122: Display location, IP of the user who have initiated the conversation
- Display location (Place, Country)
- Display IP address on the sidebar

### PR DESCRIPTION
**Pull request by sojan-official**
# Pull Request Template

## Description

- feature to store contact IP for accounts
- IP lookup through geocoder gem

Fixes #1122 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -509,6 +511,24 @@
...
+
+  create_table "workflow_account_templates", force: :cascade do |t|
+    t.string "template_id", null: false
+    t.bigint "account_id", null: false
+    t.jsonb "config", default: "{}", null: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["account_id"], name: "index_workflow_account_templates_on_account_id"
+  end
+
```
Comment (pranavrajs) [inline db/schema.rb:531]: Please remove bot flow schema changes

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please remove the bot flow changes from schema.rb

**Comment by ReactRaja** [comment]
Was this ever added?

**Comment by sojan-official** [comment]
@ReactRaja  you can find the docs to enable this on https://www.chatwoot.com/docs/self-hosted/enable-ip-logging

**Comment by ReactRaja** [comment]
Well geo2ip is paid thing hmm. 
Would geolite2 work?

**Comment by ReactRaja** [comment]
> 
> 
> Well geo2ip is paid thing hmm.
> Would geolite2 work?

So?

**Comment by sojan-official** [comment]
@ReactRaja chatwoot uses the locally downloaded MaxmindDB. It should be free to use.

**Comment by ReactRaja** [comment]
Using  `account = Account.find(account_id)` after `account_id = maxmindUSERID` seems to give an error

`ActiveRecord::RecordNotFound (Could'nt find Account with 'id'=maxmindUSERID)` 

Any idea what might be causing this?

**Comment by sojan-official** [comment]
@ReactRaja  account_id is your chatwoot account id. not your maxmind user id

**Comment by ReactRaja** [comment]
Huh so to enable for all my agents i would have to manually select their account id and enable it for them one by one? Seems a bit redundant if thats the case

**Comment by sojan-official** [comment]
@ReactRaja  account id is different from your user id.  when you load your dashboard the URL should be like this.

https://app.chatwoot.com/app/accounts/1/dashboard

1 -> is your account id

**Comment by nh2** [comment]
@sojan-official Could it be documented whether IP logging is available/configurable on the hosted https://app.chatwoot.com?

The docs seem to talk about the self-hosted version only.

**Pull request merged**